Tools/process_sensor_caldata.py

```diff
@@ -0,0 +1,821 @@
+gyro_0_params['TC_G0_TMAX'] = np.amax(sensor_gyro_0['temperature'])
+gyro_0_params['TC_G0_TREF'] = 0.5 * (gyro_0_params['TC_G0_TMIN'] + gyro_0_params['TC_G0_TMAX'])
+temp_rel = sensor_gyro_0['temperature'] - gyro_0_params['TC_G0_TREF']
+temp_rel_resample = np.linspace(gyro_0_params['TC_G0_TMIN']-gyro_0_params['TC_G0_TREF'], gyro_0_params['TC_G0_TMAX']-gyro_0_params['TC_G0_TREF'], 100)
+temp_resample = temp_rel_resample + gyro_0_params['TC_G0_TREF']
+
+# fit X axis
+coef_gyro_0_x = np.polyfit(temp_rel,-sensor_gyro_0['x'],3)
+gyro_0_params['TC_G0_X3_0'] = coef_gyro_0_x[0]
+gyro_0_params['TC_G0_X2_0'] = coef_gyro_0_x[1]
+gyro_0_params['TC_G0_X1_0'] = coef_gyro_0_x[2]
+gyro_0_params['TC_G0_X0_0'] = coef_gyro_0_x[3]
+fit_coef_gyro_0_x = np.poly1d(coef_gyro_0_x)
+gyro_0_x_resample = fit_coef_gyro_0_x(temp_rel_resample)
+
+# fit Y axis
+coef_gyro_0_y = np.polyfit(temp_rel,-sensor_gyro_0['y'],3)
+gyro_0_params['TC_G0_X3_1'] = coef_gyro_0_y[0]
+gyro_0_params['TC_G0_X2_1'] = coef_gyro_0_y[1]
+gyro_0_params['TC_G0_X1_1'] = coef_gyro_0_y[2]
+gyro_0_params['TC_G0_X0_1'] = coef_gyro_0_y[3]
+fit_coef_gyro_0_y = np.poly1d(coef_gyro_0_y)
+gyro_0_y_resample = fit_coef_gyro_0_y(temp_rel_resample)
+
+# fit Z axis
+coef_gyro_0_z = np.polyfit(temp_rel,-sensor_gyro_0['z'],3)
+gyro_0_params['TC_G0_X3_2'] = coef_gyro_0_z[0]
+gyro_0_params['TC_G0_X2_2'] = coef_gyro_0_z[1]
+gyro_0_params['TC_G0_X1_2'] = coef_gyro_0_z[2]
+gyro_0_params['TC_G0_X0_2'] = coef_gyro_0_z[3]
+fit_coef_gyro_0_z = np.poly1d(coef_gyro_0_z)
+gyro_0_z_resample = fit_coef_gyro_0_z(temp_rel_resample)
+
+# gyro0 vs temperature
+plt.figure(1,figsize=(20,13))
+
+# draw plots
+plt.subplot(3,1,1)
+plt.plot(sensor_gyro_0['temperature'],sensor_gyro_0['x'],'b')
+plt.plot(temp_resample,-gyro_0_x_resample,'r')
+plt.title('Gyro 0 Bias vs Temperature')
+plt.ylabel('X bias (rad/s)')
+plt.xlabel('temperature (degC)')
+plt.grid()
+
+# draw plots
+plt.subplot(3,1,2)
+plt.plot(sensor_gyro_0['temperature'],sensor_gyro_0['y'],'b')
+plt.plot(temp_resample,-gyro_0_y_resample,'r')
+plt.ylabel('Y bias (rad/s)')
+plt.xlabel('temperature (degC)')
+plt.grid()
+
+# draw plots
+plt.subplot(3,1,3)
+plt.plot(sensor_gyro_0['temperature'],sensor_gyro_0['z'],'b')
+plt.plot(temp_resample,-gyro_0_z_resample,'r')
+plt.ylabel('Z bias (rad/s)')
+plt.xlabel('temperature (degC)')
+plt.grid()
+
+pp.savefig()
+
+#################################################################################
+
+#################################################################################
+
+# define data dictionary of gyro 1 thermal correction  parameters
+gyro_1_params = {
+'TC_G1_ID':0,
+'TC_G1_TMIN':0.0,
+'TC_G1_TMAX':0.0,
+'TC_G1_TREF':0.0,
+'TC_G1_X0_0':0.0,
+'TC_G1_X1_0':0.0,
+'TC_G1_X2_0':0.0,
+'TC_G1_X3_0':0.0,
+'TC_G1_X0_1':0.0,
+'TC_G1_X1_1':0.0,
+'TC_G1_X2_1':0.0,
+'TC_G1_X3_1':0.0,
+'TC_G1_X0_2':0.0,
+'TC_G1_X1_2':0.0,
+'TC_G1_X2_2':0.0,
+'TC_G1_X3_2':0.0,
+'TC_G1_SCL_0':1.0,
+'TC_G1_SCL_1':1.0,
+'TC_G1_SCL_2':1.0
+}
+
```

Comment:
It would be good to compact this if we you don't want to use px4tools to make the gyro analysis a function and not copy and paste the same code.